src/js/editor/editor.js

```diff
@@ -293,7 +306,7 @@ class Editor {
     this.run(postEditor => {
       let nextPosition = postEditor.deleteAtPosition(position, direction, {unit});
       postEditor.setRange(nextPosition);
-    });
+    }, EDITOR_RUN_LAST_ACTION_STATE.DELETE);
```

Comment:
I think if it's mixed then it's unlikely to be the result of a direct manipulation by the end user and therefore should break the undo grouping. The same as some events which are insertions like a Paste event.

The reason that this works is because only two states are explicitly defined (TEXT_INSERT and DELETE which results in a grouping) and everything else is undefined which keeps them as unique elements on the undo stack.